dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -58,7 +60,8 @@ final class ConjureBodySerDe implements BodySerDe {
     private final Deserializer<Optional<InputStream>> optionalBinaryInputStreamDeserializer;
     private final Deserializer<Void> emptyBodyDeserializer;
     private final LoadingCache<Type, Serializer<?>> serializers;
-    private final LoadingCache<Type, Deserializer<?>> deserializers;
+    private final LoadingCache<Type, EncodingDeserializerForEndpointRegistry<?>> deserializers;
+    private final EmptyContainerDeserializer emptyContainerDeserializer;
```

Comment:
The `emptyContainerDeserializer` is needed to created the `EncodingDeserializerForEndpointRegistry`. Previously `EncodingDeserializerRegistry` was only created in the constructor, but not it's created in the `deserializers(DeserializerArgs<>)` method as well. I added `emptyContainerDeserializer` as a field so I can access it there.